metricbeat/mb/registry.go

```diff
@@ -116,11 +116,11 @@ type Register struct {
 type ModulesSource interface {
 	Modules() ([]string, error)
 	HasModule(module string) bool
-	MetricSets(module string) ([]string, error)
-	DefaultMetricSets(module string) ([]string, error)
+	MetricSets(r *Register, module string) ([]string, error)
+	DefaultMetricSets(r *Register, module string) ([]string, error)
```

Comment:
This is valid concern. I admit it took me a while to find the best solution for the moment (don't refactor half of the project). I'm afraid that it will be related with heavier refactoring.